subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1213,16 +1236,50 @@ int ull_adv_reset(void)
 
 inline struct ll_adv_set *ull_adv_set_get(uint8_t handle)
 {
-	if (handle >= BT_CTLR_ADV_SET) {
-		return NULL;
+	struct ll_adv_set *adv = &ll_adv[0];
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	uint8_t idx;
+
+	for (idx = 0U; idx < BT_CTLR_ADV_SET; idx++, adv++) {
+		if (adv->is_created && (adv->handle == handle)) {
+			return adv;
+		}
+	}
+
+	return NULL;
+#else
+	return adv;
+#endif
+}
+
+inline struct ll_adv_set *ull_adv_set_get_empty(void)
+{
+	struct ll_adv_set *adv = &ll_adv[0];
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	uint8_t idx;
+
+	for (idx = 0U; idx < BT_CTLR_ADV_SET; idx++, adv++) {
+		if (!adv->is_created) {
+			return adv;
+		}
 	}
 
-	return &ll_adv[handle];
+	return NULL;
+#else
+	return adv;
+#endif
 }
 
 inline uint16_t ull_adv_handle_get(struct ll_adv_set *adv)
 {
-	return ((uint8_t *)adv - (uint8_t *)ll_adv) / sizeof(*adv);
+#if defined(CONFIG_BT_CTLR_ADV_SET)
+	return adv->handle;
+#else
+	LL_ASSERT(adv == &ll_adv[0]);
+	return 0;
+#endif
```

Comment:
Zero-indexed handles is used as ticker instance offset, I think. Move HCI handle lookup as separate feature in HCI module.